docs/overview.rst

```diff
@@ -1,6 +1,22 @@
-============
-Architecture
-============
+========
+Overview
+========
```

Comment:
A while back, Will and I discussed documentation and one of the things we talked about was having an Overview doc that sets some context for every project. I added that here.